app/main.py

```diff
@@ -1 +1,29 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def check_alive(self) -> None:
+        if self.health <= 0:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, grass_animal: Herbivore) -> None:
+        if not grass_animal.hidden and isinstance(grass_animal, Herbivore):
```

Comment:
```suggestion
    def bite(self, grass_animal: Herbivore) -> None:
        if isinstance(grass_animal, Herbivore) and  not grass_animal.hidden:
```
The i`sinstance(grass_animal, Herbivore)` check should be performed before accessing `grass_animal.hidden `to ensure that `grass_animal` is indeed an instance of `Herbivore`.

Reply:
thanks